sklearn/tree/_criterion.pyx

```diff
@@ -245,25 +285,28 @@ cdef class ClassificationCriterion(Criterion):
 
         safe_realloc(&self.n_classes, n_outputs)
 
-        cdef SIZE_t k = 0
+        cdef SIZE_t output_i = 0
```

Comment:
I'm not a huge fan of this change. If anything, it adds general aesthetics, which shouldn't be forced into a functionality-based PR.